src/math/CMakeLists.txt

```diff
@@ -5,31 +5,41 @@ if(BUILD_LIBRARY)
 	return()
 endif()
 
+is_zephyr(it_is_zephyr)
+
 add_local_sources(sof numbers.c)
 
-if(CONFIG_CORDIC_FIXED)
+if(CONFIG_CORDIC_FIXED OR it_is_zephyr)
```

Comment:
Using the same name for managing math functions under Zephyr is not a good idea.

Reply:
I'm not sure I understand @ShriramShastry. The purpose of this PR is to be very careful and NOT make any config "management" change.  That can come later after this clean up. Currently, math function are NOT managed in Zephyr.

Reply:
OK, What I mean is that `it_is_zephyr` is used to enable and disable math functions. If the function is called in the code, you enable `it_is_zephyr`, otherwise you turn it off. What I'm not persuaded of is the use of the same flag (`it_is_zephyr`) across Zephyr code . If you have a solid justification (`That's Zephyr`) for using the same name, that's fine; otherwise, I advocate using a distinct naming policy for each math function.

Reply:
>  that it_is_zephyr is used to enable and disable math functions

No, that's not what it is. No one is going to "turn Zephyr on/off" to enable/disable some math functions. No one can decide whether they use Zephyr or not, it's not a choice.

Before this PR:
- CORDIC is not optional in Zephyr

After this PR:
- CORDIC is not optional in Zephyr

The purpose of this PR is to carefully NOT change that.

If you want make to make CORDIC or other math code CONFIGurable in Zephyr then please submit a separate PR. This PR is a pure clean-up, it does not make any CONFIG change.